File: 2019/03/Point.java
public class Point {
  public int x, y;

  public Point() {
    this(0, 0);
  }

  public Point(int x, int y) {
    this.x = x;
    this.y = y;
  }

  public String toString() {
    return "(" + x + "," + y + ")";
  }

  public int manhattan() {
    return Math.abs(x) + Math.abs(y);
  }
}

File: 2018/06/Point.cs
class Point {
  public int x { get; }
  public int y { get; }

  public Point(int x, int y) {
    this.x = x;
    this.y = y;
  }

  public Point(string p) {
    IList<int> c = p.Split(", ").Select(i => Int32.Parse(i)).ToList();
    this.x = c[0];
    this.y = c[1];
  }

  public int Distance(int x, int y) {
    return Math.Abs(this.x - x) + Math.Abs(this.y - y);
  }

  public override string ToString() {
    return $"({x},{y})";
  }
}
